pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -106,7 +113,7 @@ type RoleSpec struct {
 	DatabaseName string `json:"databaseName"`
 
 	// CollectionName is a collection for which the role applies.
-	CollectionName string `json:"collectionName"`
+	CollectionName string `json:"collectionName,omitempty"`
```

Comment:
while the API docs mark this parameter as required - in fact it's possible to omit it for some of the roles